qiskit_machine_learning/algorithms/classifiers/vqc.py

```diff
@@ -154,30 +163,123 @@ def circuit(self) -> QuantumCircuit:
+            y: The target values.
 
         Returns:
-            self: returns a trained classifier.
+            Trained classifier.
         """
-        num_classes = y.shape[-1]
+        y = self._validate_y(y)
+        num_classes = self._get_num_classes(y)
         cast(CircuitQNN, self._neural_network).set_interpret(
             self._get_interpret(num_classes), num_classes
         )
         return super().fit(X, y)
 
+    # pylint: disable=invalid-name
+    def score(self, X: np.ndarray, y: np.ndarray, sample_weight: np.ndarray | None = None) -> float:
+        y = self._validate_y(y)
+        return super().score(X, y, sample_weight=sample_weight)
+
     def _get_interpret(self, num_classes: int):
         def parity(x: int, num_classes: int = num_classes) -> int:
             return x % num_classes
 
         return parity
+
+    def _get_num_classes(self, y: np.ndarray) -> int:
+        """Infers the number of classes from the targets.
+
+        Args:
+            y: The target values.
+
+        Raises:
+            QiskitMachineLearningError: If the number of classes differs from
+            the previous batch when using a warm start.
+
+        Returns:
+            The number of inferred classes.
+        """
+        if self._one_hot:
+            num_classes = y.shape[-1]
+        else:
+            num_classes = sum(np.unique(y))
+
+        return self._validate_num_classes(num_classes)
+
+    def _validate_y(self, y: np.ndarray) -> np.ndarray:
+        """Transforms and validates labels.
+
+        Binarizes the labels, infers whether the result is one-hot encoded
+        and sets the relevant attribute in the parent class.
+
+        Args:
+            y: The target values
+
+        Returns:
+            The suitably transformed and validated target values.
+        """
+        targets = self._label_transformer.fit_transform(y)
+        if len(targets[0]) == 1:
+            self._one_hot = False
+            targets = targets.flatten()
+        else:
+            self._one_hot = True
+            self._validate_one_hot_targets(targets)
+        return targets
+
+    def _validate_num_classes(self, num_classes: int) -> int:
+        """Validate the number of inferred classes.
+
+        If using a warm start and the number of classes differs between batches
+        raise an error, otherwise override.
+
+        Args:
+            num_classes: The number of inferred classes.
+
+        Raises:
+            QiskitMachineLearningError: If the number of classes differs from
+            the previous batch when using a warm start.
+
+        Returns:
+            The validated number of classes.
+        """
+        if self._warm_start and self._num_classes is not None and self._num_classes != num_classes:
+            raise QiskitMachineLearningError(
+                f"Number of classes: {num_classes},\
+                differs from the previous batch: {self.num_classes}."
```

Comment:
Normally we would have the multiple strings so they end up being joined as one 
```
                f"Number of classes: {num_classes}, "
                f"differs from the previous batch: {self.num_classes}."
```
With using a line continuation does that not end up including all that leading space of the subseqent line into the final string.

Reply:
😅 sure does. Fixed.